apps/ledger-live-mobile/src/screens/Manager/Manager.tsx

```diff
@@ -47,6 +48,17 @@ const Manager = ({ navigation, route }: NavigationProps) => {
   const [state, dispatch] = useApps(result, deviceId, appsToRestore);
   const reduxDispatch = useDispatch();
 
+  const lastConnectedDevice = useSelector(lastConnectedDeviceSelector);
+  useEffect(() => {
+    // refresh the manager if a new device gets connected
+    // (happes only when we plug a new device via USB)
+    if (lastConnectedDevice?.deviceId !== device.deviceId) {
```

Comment:
which part of the code is going to setting `lastConnectedDevice` back to device? it probably should be the same place?

because when only looking at this portion of the added code, nothing show me that lastConnectedDevice is going to converge to the same device. if it's not done in the same effect, there is a risk that worse case scenario the value never is set (so the effect runs indefinitely?) or also that it is set in another cycle (so it `replace` a few times before settling 🤔 )

Reply:
We don't change `lastConnectedDevice` anywhere here, we're just reacting to a change on it (it is a global state).

This converges because we're renavigating to a previous page (this one is `ScreenName.ManagerMain`, the previous one, where we select the device, is the `ScreenName.Manager` one), with the new device, so the next time we get to this useEffect the new `device.deviceId` will be equal to `lastConnectedDevice.deviceId`.

If `lastConnectedDevice` is set to null anywhere (it shouldn't really happen in the app, but if it does), we'll just renavigate to the manager device selection screen with no device in the params, so the user will have manually to select a device to connect to.

Reply:
So, the places where we update these values when selecting a device in the manager are the following:
* https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-mobile/src/components/SelectDevice/index.tsx#L107 (for the old device selection component)
* https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-mobile/src/components/SelectDevice2/index.tsx#L160 (for the new device selection component)

On both of them we dispatch the action that updates the global state before calling the `onSelect` callback which is the callback that is navigating to this open manager screen. However, I'm not sure if this is enough to guarantee that we will have the global state up to date before entering this page 🤔. Do you have any ideas of how we could guarantee that?

Reply:
And the other place where this value is updated is on https://github.com/LedgerHQ/ledger-live/pull/3570/files#diff-7323cf8e09a7abc28bab7b04c44c296398310994e71edccdd604386682628e95
where we update it when the user plugs a new USB device, and this is the one we want to be able to react to and reload the manager.

Reply:
We update it in the connect manager flow. the big question here is if there's a way for it to not be updated by the time we declare the useEffect that reacts to it. This in theory, would cause us to enter the manager, then, right after it, reload it again because the device was updated after we entered the manager.